### ISSUE TITLE
Model fails when reliability module in place

### ISSUE THREAD

**Issue by blthayer** [gridlab-d/gridlab-d#1197]
Copy + pasting from email with @ftuffner, who has the files referred to in the email (there doesn't seem to be an easy way to attach them here).

**Initial email:**
```
I’ve attached the model that’s giving me trouble (“model_0.glm”). Here’s my version information:

GridLAB-D 4.1.0-18165 (Lugo [76964ea7:develop:Mod]) 64-bit LINUX RELEASE

And here’s the output from trying to run the attached model:

root@31479680e9eb:/tmp# gridlabd model_0.glm 
WARNING  [INIT] : object name '221-242079' does not follow strict naming rules and may not link correctly during load time
WARNING  [INIT] : last warning message was repeated 39 times
WARNING  [INIT] : Node:cap_capbank2a does not have the same nominal voltage as its parent - copying voltage from parent.
WARNING  [INIT] : last warning message was repeated 9 times
WARNING  [INIT] : Using a 2x2 z-matrix, instead of geometric values, is an under-determined system. Ground and/or neutral currents will be incorrect.
WARNING  [INIT] : last warning message was repeated 1274 times
WARNING  [INIT] : event_gen:external_event_handler does not have a metrics object as a parent -- metrics are not calculated.
WARNING  [INIT] : Node:sourcebus_meter is parented to a swing node and will get folded into it.
WARNING  [INIT] : solarpanel: no climate data found, using static data
WARNING  [2019-10-11 19:49:36 UTC] : last warning message was repeated 353 times
WARNING  [2019-10-11 19:49:36 UTC] : Regulator reg_feeder_reg1a has phase A at the maximum tap value
WARNING  [2019-10-11 19:49:36 UTC] : last warning message was repeated 17 times
WARNING  [2019-10-11 19:49:36 UTC] : Restoration object not detected!
WARNING  [2019-10-11 19:49:36 UTC] : Unsupported phase on a possibly meshed node
WARNING  [2019-10-11 19:49:36 UTC] : Empty powerflow connectivity matrix, your system is empty!
WARNING  [2019-10-11 19:49:36 UTC] : transformer:xf_hvmv115_69sub is at 2163.97% of its rated power value
WARNING  [2019-10-11 19:49:36 UTC] : Regulator reg_feeder_reg1a has phase A at the maximum tap value
WARNING  [2019-10-11 19:49:36 UTC] : last warning message was repeated 17 times
WARNING  [2019-10-11 19:49:36 UTC] : Empty powerflow connectivity matrix, your system is empty!
WARNING  [2019-10-11 19:49:36 UTC] : Line:line_hvmv_sub_hsb is at 361024.20% of its emergency rating on phase A!
WARNING  [2019-10-11 19:49:36 UTC] : Line:line_hvmv_sub_hsb is at 361024.20% of its emergency rating on phase B!
WARNING  [2019-10-11 19:49:36 UTC] : Line:line_hvmv_sub_hsb is at 361024.20% of its emergency rating on phase C!

I suspect this is related to the reliability module, fault_check object, and eventgen object. I’m not trying to actually calculate any metrics, but Andy told me that I needed these objects to make the model run at all (see here). I believe this has something to do with the system being “topologically meshed but operationally radial.”

What I believe is the most concerning warning is “Empty powerflow connectivity matrix, your system is empty!” What’s bizarre to me is that there are warnings about overloads, but it seems like the system isn’t actually “running” (I get no results published to my MySQL database when the MySQL related objects are not commented-out. However, if GridLAB-D cannot connect to the MySQL database I believe it throws an error, so I’m pretty confident there isn’t a database connectivity issue).

For comparison, I’ve also attached a zipped folder (1613025849.zip) of all the files the GridAPPS-D platform uses to run a similar model. The files of note are “model_base.glm” which contains just the grid-related objects, and “model_startup.glm” which includes “model_base.glm” and has things like the clock and modules. When I inspect “model_startup.glm,” it seems like I’m including everything I should be in my model (“model_0.glm”). The power system components should be identical between the two models.

I was able to get the “model_startup.glm” model running after commenting out all related FNCS stuff and making the “#include ” line use relative pathing. It doesn’t seem to suffer from the same error. 

I tried moving my “eventgen” and “fault_check” objects to earlier in the model (I know sometimes GridLAB-D cares about object ordering), but that didn’t seem to help.

I’m not sure how to start debugging this from here.
```

**Response:**
```
The main file (just the bare GLM) seems to be iterating on the same timestep a lot.  The system eventually ends up disconnecting everything except the SWING node at “sourcebus”, that’s the source of the “Empty matrix” messages, because it is.  I have no idea what is causing that though, since there are no external event drivers and nothing changing the system topology (so far as I could tell).  The files appear pretty similar (roughly the same object count, plus the XMLs don’t show any massive swath of different objects).

I’ll loop back with you on Monday, unless I have an epiphany over the weekend.  At this point, I’m a perplexed as you are.  My next inclination would be to look at the admittance matrix, but it looks like things get loaded in different orders (so the entries wouldn’t line up and would make debugging hard).  Given how the “standalone file” iterates a lot on the first timestep before failing, I’m wondering what is causing that.

If you have any luck/ideas, let me know!
```

**Re: re:**
```
Well, I found the issue (though I certainly don’t understand why it’s an issue, and would appreciate any insight you may have).

In my model (model_0.glm), I added a meter to the swing node, and have the line coming out of the swing node originate “from” the meter, rather than the substation. This has worked fine on other models (e.g. IEEE 8500), so I’m not sure why it’s failing here. 

Here are the relevant snippets from model_0.glm (not in order, but that shouldn’t matter):

object substation {
                name "sourcebus";
                bustype SWING;
                phases ABCN;
                nominal_voltage 66395.28;
                base_power 12MVA;
                power_convergence_value 100VA;
                positive_sequence_voltage ${VSOURCE};
};

object meter {
                phases ABCN;
                nominal_voltage 66395.28;
                parent "sourcebus";
                name "sourcebus_meter";
};

object overhead_line {
                name "line_hvmv_sub_hsb";
                from "sourcebus_meter";
                to "hvmv115_hsb1";
                phases ABC;
                length 3.2809;
                continuous_rating_A 400.00;
                continuous_rating_B 400.00;
                continuous_rating_C 400.00;
                emergency_rating_A 400.00;
                emergency_rating_B 400.00;
                emergency_rating_C 400.00;
                configuration "lcon_hvmv_sub_hsb_ABC";
};

And here are the components from “model_base.glm”:

object overhead_line {
  name "line_hvmv_sub_hsb";
  from "sourcebus";
  to "hvmv115_hsb1";
  phases ABC;
  length 3.2809;
  continuous_rating_A 400.00;
  continuous_rating_B 400.00;
  continuous_rating_C 400.00;
  emergency_rating_A 400.00;
  emergency_rating_B 400.00;
  emergency_rating_C 400.00;
  configuration "lcon_hvmv_sub_hsb_ABC";
}

object substation {
  name "sourcebus";
  bustype SWING;
  phases ABCN;
  nominal_voltage 66395.28;
  base_power 12MVA;
  power_convergence_value 100VA;
  positive_sequence_voltage ${VSOURCE};
}


When I change “model_0.glm” to look like “model_base.glm” (by commenting out the meter, and changing the “from” field on the overhead line), “model_0.glm” runs without the “your system is empty” issue. 
```

**Re: re: re:**
```
I had noticed that, but didn’t think anything of it – what you did *SHOULD* work just the same, but obviously isn’t.  That at least narrows it down – there may be an issue with how the “parent reassignment” is being done.  When you have lines connected to childed nodes, it effectively “moves” them inside the admittance matrix to the parent node (and then adjust any current injections at the end).  This is mainly due to how we form our admittance – it wouldn’t like “zero impedance links” between the parent and child.  Given you’re parenting it to a substation, it may be something as stupid/silly as it not looking for that object type (and half-failing, as a result).

My initial guess is the topology checker is getting confused by this, though I’m not sure how (it references an internal NR structure, and should be ignorant to the childed node).  Something to try real quick, if you have a chance (and probably the first thing I’ll try), is to replace the substation object with just a node or a normal meter, and see if that works.  I have a suspicion it probably will, and there’s just a logic hole I need to plug somewhere.

I won’t have a chance to poke around with this until Monday, but at least this gives us a starting point to debug it!
```

**Re: re: re: re:**
```
Thanks for getting back to me! I circumvented the issue by no longer setting the line’s “from” parameter to reference the meter. When I set my code up to change the line’s “from,” I did it to be extra safe because I thought there was a chance the meter wouldn’t register the flow through it unless the line was “from” the meter, rather than the parented substation object. However, I did some tests, and the meter object does still properly record flow if it’s parented to the substation object and the line is “from” the substation object.
```

**Comment by ftuffner** [comment]
Issue has to deal with out-of-sequence initialization, especially with SWING-based nodes.  If the child is before the initializing line, it works fine.  Figuring out how to fix this - may be as simple as deferred inits for all lines until the nodes have initialized.

**Comment by ftuffner** [comment]
Merged as PR #1199 - closing
